diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,7 @@ set( CMAKE_OSX_DEPLOYMENT_TARGET "10.9" )
 # Universal builds for mac
 # set( CMAKE_OSX_ARCHITECTURES "i386;x86_64" )
 
+find_package( ZLIB )
 
 # The parsing order is significant!
 
@@ -61,7 +62,10 @@ add_subdirectory( src/XercesExtensions )
 set( XERCESEXTENSIONS_LIBRARIES XercesExtensions )
 set( XERCESEXTENSIONS_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/src/XercesExtensions )
 
-add_subdirectory( src/zlib )
+if( NOT ZLIB_FOUND )
+    add_subdirectory( src/zlib )
+endif()
+
 add_subdirectory( src/zipios )
 if( NOT MINGW )
     add_subdirectory( src/googlemock )
diff --git a/src/zipios/CMakeLists.txt b/src/zipios/CMakeLists.txt
--- a/src/zipios/CMakeLists.txt
+++ b/src/zipios/CMakeLists.txt
@@ -17,15 +17,25 @@ file( GLOB_RECURSE SOURCES *.cpp *.h )
 # and the headers for the linked-to libraries
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR}
                      ${BoostParts_SOURCE_DIR}
-                     ${zlib_SOURCE_DIR}
-                     ${zlib_BINARY_DIR}
                      )
 
+if( ZLIB_FOUND )
+    include_directories( ${ZLIB_INCLUDE_DIRS} )
+else()
+    include_directories( ${zlib_SOURCE_DIR} ${zlib_BINARY_DIR} )
+endif()
+
 link_directories ( ${PROJECT_BINARY_DIR}/lib )
 
 add_library( ${PROJECT_NAME} STATIC ${SOURCES} )
 
-target_link_libraries( ${PROJECT_NAME} zlib BoostParts )
+target_link_libraries( ${PROJECT_NAME} BoostParts )
+
+if( ZLIB_FOUND )
+    target_link_libraries( ${PROJECT_NAME} ${ZLIB_LIBRARIES} )
+else()
+    target_link_libraries( ${PROJECT_NAME} zlib )
+endif()
 
 #############################################################################
 
