From fcfc63fa7d51b54c7144d160dbc360503b61acc0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan[at]danny.cz>
Date: Wed, 18 Jan 2012 20:40:27 +0100
Subject: [PATCH 05/12] fix build with boost >= 1.48

---
 src/FlightCrew/Validators/Opf/ReachabilityAnalysis.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/FlightCrew/Validators/Opf/ReachabilityAnalysis.cpp b/src/FlightCrew/Validators/Opf/ReachabilityAnalysis.cpp
index 24c0f28..35218e1 100644
--- a/src/FlightCrew/Validators/Opf/ReachabilityAnalysis.cpp
+++ b/src/FlightCrew/Validators/Opf/ReachabilityAnalysis.cpp
@@ -26,7 +26,9 @@
 #include <XmlUtils.h>
 #include "Misc/DetermineMimetype.h"
 #include "Misc/Utilities.h"
+#include <boost/version.hpp>
 
+#if BOOST_VERSION < 104800
 namespace boost
 {
 
@@ -48,6 +50,7 @@ namespace filesystem3
 } // namespace filesystem3
 
 } // namespace boost
+#endif
 
 
 namespace FlightCrew
-- 
1.8.5.3

