From 1ff4c1b395f7eda71191cda1c010ae5cb24ea6ff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan[at]danny.cz>
Date: Wed, 18 Jan 2012 20:42:27 +0100
Subject: [PATCH 08/12] don't build googlemock when NO_TEST_EXE is specified

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 872ec34..301ac38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,9 @@ if( NOT ZLIB_FOUND )
     add_subdirectory( src/zlib )
 endif()
 add_subdirectory( src/zipios )
-add_subdirectory( src/googlemock )
+if( NOT NO_TEST_EXE )
+    add_subdirectory( src/googlemock )
+endif()
 add_subdirectory( src/FlightCrew )
 add_subdirectory( src/FlightCrew-cli )
 
-- 
1.8.5.3

