From 2973b6c50c86b4477be3b1fd3399c62bac0d9497 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 18 Jan 2012 20:44:47 +0100
Subject: [PATCH 11/11] allow building zipios as a shared lib

---
 src/zipios/CMakeLists.txt |   23 ++++++++++++++++++++---
 1 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/src/zipios/CMakeLists.txt b/src/zipios/CMakeLists.txt
index 3ab775c..781c30d 100644
--- a/src/zipios/CMakeLists.txt
+++ b/src/zipios/CMakeLists.txt
@@ -27,7 +27,19 @@ endif()
 
 link_directories ( ${PROJECT_BINARY_DIR}/lib )
 
-add_library( ${PROJECT_NAME} ${SOURCES} )
+if( BUILD_SHARED_ZI )
+    add_library( ${PROJECT_NAME} SHARED ${SOURCES} )
+    set_target_properties( ${PROJECT_NAME} PROPERTIES
+        VERSION "${FLIGHTCREW_LIBVERSION}"
+        SOVERSION "${FLIGHTCREW_SOVERSION}"
+        OUTPUT_NAME "${PROJECT_NAME}"
+    )
+else()
+    add_library( ${PROJECT_NAME} ${SOURCES} )
+    if( UNIX AND BUILD_SHARED_FC )
+        add_definitions(-fPIC)
+    endif()
+endif()
 
 target_link_libraries( ${PROJECT_NAME} ${BOOST_LIBS} )
 if( ZLIB_FOUND )
@@ -57,6 +69,11 @@ elseif( CMAKE_COMPILER_IS_GNUCXX )
     add_definitions( -Wall )
 endif()
 
-if( UNIX AND BUILD_SHARED_FC )
-    add_definitions(-fPIC)
+if( BUILD_SHARED_ZI AND UNIX AND NOT APPLE )
+    install( TARGETS ${PROJECT_NAME}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+     )
+    file( GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/zipios++/*.h" )
+    install( FILES ${files} DESTINATION ${INCLUDE_INSTALL_DIR}/${PROJECT_NAME}/zipios++ )
 endif()
-- 
1.7.7.4

